tests/Platforms/AbstractMySQLPlatformTestCase.php

```diff
@@ -808,6 +807,108 @@ protected function getQuotedAlterTableChangeColumnLengthSQL(): array
         ];
     }
 
+    public function testAlterTableChangeBlobColumnLength(): void
+    {
+        $fromTable = new Table('mytable');
+
+        $fromTable->addColumn('blob1', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_TINYBLOB]);
+        $fromTable->addColumn('blob2', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_BLOB]);
+        $fromTable->addColumn('blob3', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB]);
+        $fromTable->addColumn('blob4', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB * 2]);
+        $fromTable->addColumn('blob5', 'blob');
+        $fromTable->addColumn('blob6', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_TINYBLOB]);
+        $fromTable->addColumn('blob7', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_BLOB]);
+        $fromTable->addColumn('blob8', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB]);
+        $fromTable->addColumn('blob9', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB * 2]);
+        $fromTable->addColumn('blob10', 'blob');
+
+        $toTable = new Table('mytable');
+
+        $toTable->addColumn('blob1', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_BLOB]);
+        $toTable->addColumn('blob2', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB]);
+        $toTable->addColumn('blob3', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB * 2]);
+        $toTable->addColumn('blob4', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_TINYBLOB]);
+        $toTable->addColumn('blob5', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_BLOB]);
+        $toTable->addColumn('blob6', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_TINYBLOB]);
+        $toTable->addColumn('blob7', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_BLOB]);
+        $toTable->addColumn('blob8', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB]);
+        $toTable->addColumn('blob9', 'blob', ['length' => MySQLPlatform::LENGTH_LIMIT_MEDIUMBLOB * 2]);
+        $toTable->addColumn('blob10', 'blob');
+
+        $diff = (new Comparator())->diffTable($fromTable, $toTable, $this->platform);
```

Comment:
It doesn't look like a platform test should involve the comparator. The only responsibility of the platform is to generate DDL.

Reply:
This is already the case in https://github.com/doctrine/dbal/blob/25496429ef4bba2352b0ced7daffe1b3dec83d13/tests/Platforms/AbstractPlatformTestCase.php#L965
We want to test if the comparator works correctly with every platform, so I think it is the right place here.